### ISSUE TITLE
Store a history of plugin installs/activations/upgrades

### ISSUE THREAD

**Issue by jdeeburke** [skyverge/wc-plugin-framework#314]
It would be really handy for support debugging purposes if we had a reliable history of events for when a plugin was installed/uninstalled, activated/deactivated, or most importantly, upgraded.

See [Slack Convo](https://skyverge.slack.com/archives/C02FLMBUR/p1534791489000100)

This would be stored in the options table in a format something like this:
```js
[
    <timestamp>: {
        action: "install|upgrade|deactivate|activate|uninstall",
        version: "plugin_version",
        previous_version: "previous_version", // only if upgrading
    },
]
```

**Comment by unfulvio** [comment]
I dig this. At the same time can we include some handling of the plugin upgrade that follows this convention we have adopted in a number of plugins?

https://github.com/skyverge/woocommerce-memberships/blob/b7c0daa48aa5e96ec5818bcefb13dabcd9a6fe30/includes/class-wc-memberships-upgrade.php#L172-L197

(linking Memberships here but you'll find this in a bunch of plugins)

We could provide a helper in the Lifecycle handler to loop through a similar array that can be populated by the plugin implementation.

On each step it writes a note to log about begin/end upgrade. So we can tell if the upgrade ran, or if it was interrupted etc. Individual upgrade methods for each version step can then add more logging, if necessary.

I can prepare a PR later if you like this.

**Comment by lkraav** [comment]
I'm rooting for more https://github.com/xwp/stream integrations, but not sure if that works here in the big picture type of way.

**Comment by ChaseWiseman** [comment]
@unfulvio yeah I already implemented something similar on this branch, see https://github.com/skyverge/wc-plugin-framework/pull/334/commits/98fe33dcbbdf13cec7095180bacfbde94a4ff3bf

That way all a plugin needs to do is add a version to the array, and a corresponding `update_to_x_x_x` method to the handler. This accounts for dev-versioned plugins as well.

I also added the event data to the `system_status` REST API response by default.
